File: CropDetailsUserInterface/src/app/Models/BankDetails.ts
export interface BankDetails{
    accountId :number,
    accountNumber :string,
    bankName :string,
    ifscCode : string,
    userId : number, 
    // user  : any
}

export interface BankDetailsDTO{
    accountNumber :string,
    bankName :string,
    ifscCode : string,
    userId : number
}

File: Crop Deal/Crop Deal/Model/Domain/BankDetails.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Numerics;

namespace Crop_Deal.Model.Domain
{
    public class BankDetails
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int AccountId { get; set; }
        [Required]
        public string AccountNumber { get; set; }
        [Required]
        public string BankName { get; set; }
        [Required]
        public string IfscCode { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }
        public User Users { get; set; }
    }
}
